### PR TITLE
feat: support opentelemetry metrics

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at fd11b09</samp>

This pull request adds a new `emqx_opentelemetry` application that integrates OpenTelemetry with EMQ X for collecting and exporting metrics. It also refactors and updates some existing modules, such as `emqx_prometheus`, `emqx_stats`, and `emqx_mgmt`, to improve metric naming, configuration, and collection.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt.erl [hunk 1]
```diff
@@ -185,22 +186,33 @@ stopped_node_info(Node) ->
...
-            %% Not support for Windows
-            {_, 0, 0, _} -> 0;
-            {_Num, _Use, IdleList, _} -> proplists:get_value(idle, IdleList, 0)
-        end,
-    RunQueue = erlang:statistics(run_queue),
+    Idle = vm_stats('cpu.idle'),
     {MemUsedRatio, MemTotal} = get_sys_memory(),
     [
-        {run_queue, RunQueue},
+        {run_queue, vm_stats('run.queue')},
```
Comment (savonarola) [inline apps/emqx_management/src/emqx_mgmt.erl:192]: Curious: why don't we use the same atom format here, `'run.queue'`? We convert to it below.

File: apps/emqx_opentelemetry/src/emqx_otel.erl [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    create_metric_views();
+setup(_Conf) ->
+    cleanup(),
+    ok.
+
+ensure_apps(Conf) ->
+    #{exporter := #{interval := ExporterInterval}} = Conf,
+    {ok, _} = application:ensure_all_started(opentelemetry_exporter),
+    _ = application:stop(opentelemetry_experimental),
+    ok = application:set_env(
```
Comment (savonarola) [inline apps/emqx_opentelemetry/src/emqx_otel.erl:59]: Curious: why don't we set/update this with hocon like `opentelemetry_exporter` ?

File: apps/emqx_management/src/emqx_mgmt.erl (hunk 1 above)
Comment (zhongwencool) [reply apps/emqx_management/src/emqx_mgmt.erl:192]: OpenTelemetry 's metric name  is format as `'x.y.z'` not `x_y_z`.
Prometheus's metrics name is format as `x_y_z`

Open telemetry Exporter will call vm_stats('run.queue') when collecting.
Prometheus will call vm_stats() when collecting.

So I make vm_stats/0 to call vm_stats/1

File: apps/emqx_opentelemetry/src/emqx_otel.erl (hunk 2 above)
Comment (zhongwencool) [reply apps/emqx_opentelemetry/src/emqx_otel.erl:59]: You mean to directly mapping the interval to the env in the schema, right?
```
{"interval",
            ?HOCON(
                emqx_schema:timeout_duration_ms(),
                #{
                    default => <<"10s">>,
                    mapping => "opentelemetry_exporter.interval", // or "opentelemetry_experimental.reader"
                    required => true,
                    desc => ?DESC(interval)
                }
            )}
```
the `opentelemetry_experimental.reader` is a `[map()]` type, We need add convertor for this.
Also the interval is not only for metrics reader, I plan to use it as trace/log's interval too. 
So I don't what to mapping interval as single env.

**Review by HJianBo — APPROVED** [review]

File: rebar.config [hunk 3]
```diff
@@ -84,6 +84,14 @@
     %% in conflict by erlavro and rocketmq
     , {jsone, {git, "https://github.com/emqx/jsone.git", {tag, "1.7.1"}}}
     , {uuid, {git, "https://github.com/okeuday/uuid.git", {tag, "v2.0.6"}}}
+%% trace
+      , {opentelemetry_api, {git_subdir, "http://github.com/emqx/opentelemetry-erlang", {branch, "main"}, "apps/opentelemetry_api"}}
```
Comment (zmstone) [inline rebar.config:88]: pin a tag.

**Pull request merged**